File: Cpp/Algorithm/src/GSSY/SlideWindow/MaxArea_11.cpp
#include "../gssy.h"
using namespace GSSY;
int MaxArea_11::maxArea(vector<int>& height) {
	int ans = 0;
	int len = height.size() - 1;
	for (int l = 0, r = len - 1; l < r;) {
		int h = height[l] < height[r] ? height[l++] : height[r--];
		int area = (r - l + 1) * h;
		ans = max(ans, area);
	}
	return ans;
}

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/TopInterview2Round/maxArea_11.java
package algo.tzashinorpu.SpecialSubject.TopInterview2Round;

public class maxArea_11 {
    public int maxArea(int[] height) {
        int res = 0;
        for (int l = 0, r = height.length - 1; l < r; ) {
            int h = height[l] < height[r] ? height[l++] : height[r--];
            res = Math.max(res, (r - l + 1) * h);
        }
        return res;
    }
}
